src/style/ansi.rs

```diff
@@ -78,43 +78,256 @@ impl fmt::Display for Colored {
     }
 }
 
+/// Utility function for ANSI parsing in Color and Colored.
+/// Gets the next element of `iter` and tries to parse it as a u8.
+fn parse_next_u8<'a>(iter: &mut impl Iterator<Item = &'a str>) -> Option<u8> {
+    iter.next()
+        .and_then(|s| u8::from_str_radix(s, 10).map(Some).unwrap_or(None))
+}
+
+impl Colored {
+    /// Parse an ANSI foreground or background color.
+    /// This is the string that would appear within an `ESC [ <str> m` escape sequence, as found in
+    /// various configuration files.
+    ///
+    /// For example: `38;5;0 -> ForegroundColor(Black)`,
+    ///              `38;5;26 -> ForegroundColor(AnsiValue(26))`
+    ///              `48;2;50;60;70 -> BackgroundColor(Rgb(50, 60, 70))`
+    ///              `49 -> BackgroundColor(Reset)`
+    /// Invalid sequences map to None.
+    ///
+    /// Currently, 3/4 bit color values aren't supported so return None.
+    ///
+    /// See also: [Color::parse_ansi](enum.Color.html#method.parse_ansi)
+    pub fn parse_ansi(ansi: &str) -> Option<Self> {
+        use Colored::{BackgroundColor as BG, ForegroundColor as FG};
```

Comment:
I would just leave this away and use them inline. I find it confusing if types are renamed makes code a bit more difficult to read imo.

Reply:
Thanks :). I've changed it to not rename them. I've left the `use Colored::{BackgroundColor, ForegroundColor}` in, to keep the lines where they're used shorter. I think they look quite cumbersome with the fully qualified names. I'm happy to remove the use statement if you disagree though.